utils/run_helpers.py

```diff
@@ -1,11 +1,46 @@
-from type import Experiment
+from type import Experiment, StagingConfig
 from blocks.pipeline import Pipeline
 from configs.constants import Const
 from .flatten import flatten
 from blocks.base import DataSource
 from type import Hierarchy
 
 
+def overwrite_preprocessing_configs_(
+    experiment: Experiment, staging_config: StagingConfig
+) -> None:
+    """
+    Takes global config values and overwrites the preprocessing config according to the logic of the parameters
+
+    Parameters
+    ----------
+    config
+        Configurations of the experiment (global settings of the experiment/run)
+    stagingConfig
+        The global stagingconfiguration
+
+    Returns
+    -------
+    None
+
+    """
+
+    if (
+        hasattr(staging_config, "limit_dataset_to")
+        and staging_config.limit_dataset_to is not None
+    ):
+        experiment.preprocessing_config.test_size = staging_config.limit_dataset_to
```

Comment:
@szemyd : sooo! unfortunately this happens later, after we have already "processed" preprocessing_config. have a look at the order of execution! 
we can have an additional step here again, where for any dataset (it's a Dictionary of datsets right now) we "cut" it. but modifying the preprocessing config here does nothing, as it happens sooner.